### PR TITLE
Request to merge Feature/join-table into dev

### PR DESCRIPTION
**Pull request by flemton**
Dear reviewer, in this PR I made the following commits with screenshots:

- [Create a table named vets](https://github.com/flemton/vet-clinic/commit/8fe85588a2f60f3f6a3def668a3b4a147b79e479)
Screenshot 2023-06-17 at 11 45 10 PM

- [Create a "join table" called specializations](https://github.com/flemton/vet-clinic/commit/9452cc0b42f97c846d8c21c8e1245001356a66ee)
Screenshot 2023-06-18 at 12 18 34 AM

- [Create a "join table" called visits](https://github.com/flemton/vet-clinic/commit/8bfe5c70fe93494ae3dc8a73b1240e7b9bba4c54)
Screenshot 2023-06-18 at 12 19 26 AM

- [Enter data for vets](https://github.com/flemton/vet-clinic/commit/53661aeb767b46f56f2dec406733fd63132f98fa)
Screenshot 2023-06-18 at 12 21 37 AM

- [Enter data for specialization](https://github.com/flemton/vet-clinic/commit/3b4a0ae4878d73533944517a2b29bd2325e3322c)
Screenshot 2023-06-18 at 12 24 40 AM

- [Enter data for visits](https://github.com/flemton/vet-clinic/commit/5fd4684c37a6b3a9b62c5c8f683ccc300f5241c9)
Screenshot 2023-06-18 at 12 26 46 AM

- [Used queries to answer questions](https://github.com/flemton/vet-clinic/commit/1c2f8071a0be8de7a92aa1249f4b91e0c07dfad3)
Screenshot 2023-06-18 at 1 47 37 AM
Screenshot 2023-06-18 at 12 29 19 AM
Screenshot 2023-06-18 at 12 31 27 AM
Screenshot 2023-06-18 at 12 32 50 AM
Screenshot 2023-06-18 at 1 50 30 AM
Screenshot 2023-06-18 at 12 33 39 AM
Screenshot 2023-06-18 at 12 34 26 AM
Screenshot 2023-06-18 at 12 36 33 AM
Screenshot 2023-06-18 at 12 37 26 AM

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -81,4 +81,59 @@ WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
 
 SELECT owners.full_name, COUNT(animals.name) FROM animals 
 JOIN owners ON animals.owner_id = owners.id 
-GROUP BY (owners.full_name);
\ No newline at end of file
+GROUP BY (owners.full_name);
+
+SELECT visits.date_of_visit, animals.name, vets.name FROM visits
```
Comment (Lucas-Masaba) [inline queries.sql:86]: - [x] Here, you are supposed to single out the last animal that William Thatcher saw. In this query,  you just listed animals from last date to the current date. Please fix that. You can use `DESC
  LIMIT 1;
` at the end of this query before the semi-colons

File: queries.sql [hunk 2]
```diff
@@ -81,4 +81,59 @@ WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
...
+JOIN species ON specialization.species_id = species.id
+RIGHT JOIN vets ON specialization.vet_id = vets.id;
+
+SELECT animals.id, vets.name, animals.name, date_of_visit FROM visits
+JOIN vets ON visits.vet_id = vets.id
+JOIN animals ON visits.animal_id = animals.id
+WHERE vets.name = 'Stephanie Mendez' 
+AND date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
+
+SELECT animal_id, animals.name, COUNT(animal_id) FROM visits
```
Comment (Lucas-Masaba) [inline queries.sql:108]: - [x] Here, you are supposed to single out that one animal that has the most visits to vets. Here you are listing animals in ascending order. Please fix this.

File: queries.sql [hunk 3]
```diff
@@ -81,4 +81,59 @@ WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
...
+JOIN animals ON visits.animal_id = animals.id
+GROUP BY animal_id, animals.name
+order by COUNT(animal_id);
+
+SELECT animal_id, animals.name, vets.name AS vet_name, date_of_visit FROM visits
```
Comment (Lucas-Masaba) [inline queries.sql:114]: - [x] Here, you are supposed to single out that one animal that visited Maisy Smith first. You instead listed out each animal that visited Maisy Smith in ascending order. Here you can use the command `ASC
  LIMIT 1;
` at the end of the query to fix that :+1:

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @flemton 👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- All SQL is in its respective files :file_cabinet: 
- Professional README :scroll: 
- Well written pull request message :writing_hand: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by flemton** [comment]
Dear @Lucas-Masaba, thank you for the  thorough review, suggested changes were applied and screenshots updated, cheers 🥂.

**Review by franclobo — APPROVED** [review]
Hi @flemton, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Correct queries ✔️ 
- Correct schema ✔️ 
- Correct data ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**